docs/examples/tls_mutual_authentication.rst

```diff
@@ -1,60 +1,44 @@
-SSL due to security vulnerabilities.
-
-See https://www.rabbitmq.com/ssl.html for certificate creation and rabbitmq SSL configuration instructions.
+This example demonstrates a TLS session with RabbitMQ using mutual authentication (server and client authentication). It was tested against RabbitMQ 3.7.4, using Python 3.6.5 and Pika 1.0.0b1.
 
+See https://www.rabbitmq.com/ssl.html for certificate generation and RabbitMQ TLS configuration.
 
 tls_example.py::
 
-    import ssl
-    import pika
     import logging
+    import pika
+    import ssl
 
     logging.basicConfig(level=logging.INFO)
-
-    context = ssl.SSLContext(ssl.PROTOCOL_TLSv1)
-    context.verify_mode = ssl.CERT_REQUIRED
-    context.load_verify_locations('/Users/me/tls-gen/basic/testca/cacert.pem')
-    context.load_cert_chain('/Users/me/tls-gen/basic/client/cert.pem',
-                            '/Users/me/tls-gen/basic/client/key.pem')
-
-    cp = pika.ConnectionParameters(ssl_options=pika.SSLOptions(context))
-
-    conn = pika.BlockingConnection(cp)
-    ch = conn.channel()
-    print(ch.queue_declare("sslq"))
-    ch.publish("", "sslq", "abc")
-    print(ch.basic_get("sslq"))
-
+    context = ssl.create_default_context(
+        cafile="/Users/me/tls-gen/basic/testca/cacert.pem")
+    context.load_cert_chain("/Users/me/tls-gen/basic/client/cert.pem",
+                            "/Users/me/tls-gen/basic/client/key.pem")
+    ssl_options = pika.SSLOptions(context, "example.com")
+    conn_params = pika.ConnectionParameters(port=5671,
+                                            ssl_options=ssl_options)
+    
+    with pika.BlockingConnection(conn_params) as conn:
+        ch = conn.channel()
+        ch.queue_declare("foobar")
+        ch.publish("", "foobar", "Hello, world!")
+        print(ch.basic_get("foobar"))
 
 rabbitmq.config::
 
-    %% Both the client and rabbitmq server were running on the same machine, a MacBookPro laptop.
-    %%
-    %% rabbitmq.config was created in its default location for OS X: /usr/local/etc/rabbitmq/rabbitmq.config.
-    %%
-    %% The contents of the example rabbitmq.config are for demonstration purposes only. See https://www.rabbitmq.com/ssl.html for instructions about creating the test certificates and the contents of rabbitmq.config.
-
-
-    [
-      {rabbit,
-        [
-          {ssl_listeners, [{"127.0.0.1", 5671}]},
-
-          %% Configuring SSL.
-          %% See http://www.rabbitmq.com/ssl.html for full documentation.
-          %%
-          {ssl_options, [{cacertfile,           "/Users/me/tls-gen/basic/testca/cacert.pem"},
-                         {certfile,             "/Users/me/tls-gen/basic/server/cert.pem"},
-                         {keyfile,              "/Users/me/tls-gen/basic/server/key.pem"},
-                         {verify,               verify_peer},
-                         {fail_if_no_peer_cert, true}]}
-        ]
-      }
-    ].
+    # Enable A.M.Q.P.S.
```

Comment:
Please make sure these are consistent with https://github.com/pika/pika/blob/master/testdata/rabbitmq.conf.in

Reply:
@maggyero, sorry, I should have been more specific - I meant the properties being set, not the values (e.g., file paths) themselves. Thank you.

Reply:
To me the other properties in the suggested rabbitmq.conf.in (``listeners.tcp.default = 5672``, ``num_acceptors.tcp = 10``, ``num_acceptors.ssl = 10``, ``reverse_dns_lookups = false``, ``loopback_users.guest = true``, ``log.console = false``, ``log.console.level = debug``) are distracting for the reader who would expect in an example like this only the necessary parameters for setting up TLS.

Reply:
I agree with @maggyero.